bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/providers/models/SonyThingChannelDefinition.java

```diff
@@ -0,0 +1,126 @@
+ * information.
+ * 
+ * Note that this can be either created via the constructor (which will enforce null/empty checks) or via
+ * deserialization (which can provide null/empty variables)
+ *
+ * @author Tim Roberts - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class SonyThingChannelDefinition {
+    /** The channel identifier */
+    private @Nullable final String channelId;
+
+    /** The mapped channel identifier (null if no mapping) */
+    private @Nullable final String mappedChannelId;
+
+    /** The channel type identifier */
+    private @Nullable final String channelType;
+
+    /** The channel properties */
+    private @Nullable final Map<@Nullable String, @Nullable String> properties;
+
+    /** The channel options */
+    private @Nullable final SonyThingStateDefinition state;
+
+    /**
+     * Constructs the definition from the passed arguments.
+     *
+     * @param channelId the non-null, non-empty channel identifier
+     * @param mappedChannelId the possibly null, possibly empty mapped channel identifier
+     * @param channelType the non-null, non-empty channel type
+     * @param state the non-null thing state definition
+     * @param properties the non-null, possibly empty properties
+     */
+    public SonyThingChannelDefinition(final String channelId, @Nullable final String mappedChannelId,
+            final String channelType, final SonyThingStateDefinition state, final Map<String, String> properties) {
+        Validate.notEmpty(channelId, "channelId must not be empty");
+        Validate.notEmpty(channelType, "channelType must not be empty");
+        Objects.requireNonNull(state, "state cannot be null");
+        Objects.requireNonNull(properties, "properties cannot be null");
+
+        this.channelId = channelId;
+        this.mappedChannelId = mappedChannelId;
+        this.channelType = channelType;
+        this.properties = new HashMap<>(properties);
+        this.state = state;
+    }
+
+    /**
+     * Returns the channel identifier
+     *
+     * @return a possibly null, possibly empty channel identifier
+     */
+    public @Nullable String getChannelId() {
+        return channelId;
+    }
+
+    /**
+     * Returns the mapped channel identifier
+     *
+     * @return a possibly null, never empty (if not null) mapped channel identifier
+     */
+    public @Nullable String getMappedChannelId() {
+        return StringUtils.defaultIfEmpty(mappedChannelId, null);
+    }
+
+    /**
+     * Returns the channel type identifier
+     *
+     * @return a possibly null, possibly empty channel type identifier
+     */
+    public @Nullable String getChannelType() {
+        return channelType;
+    }
+
+    /**
+     * Returns a new properties map
+     *
+     * @return a non-null, possibly empty map of properties
+     */
+    public Map<@Nullable String, @Nullable String> getProperties() {
+        final Map<@Nullable String, @Nullable String> localProp = properties;
```

Comment:
nullable keys? in a properties map?

Reply:
Class is used for deserialization - so a null key could happen